src/renderer/sw_engine/tvgSwFill.cpp

```diff
@@ -201,7 +202,17 @@ bool _prepareRadial(SwFill* fill, const RadialGradient* radial, const Matrix* tr
         fill->radial.fy = cy + r * (fy - cy) / dist;
         fill->radial.dx = cx - fill->radial.fx;
         fill->radial.dy = cy - fill->radial.fy;
+#if TVG_FLOAT_CONTRACT_AVAILABLE
         fill->radial.a = fill->radial.dr * fill->radial.dr - fill->radial.dx * fill->radial.dx - fill->radial.dy * fill->radial.dy;
+#else
+        // Prevent loss of precision on Apple Silicon when dr=dy and dx=0 due to FMA
```

Comment:
I think we can replace the new logic with the old one. Maintaining both logics just increases the maintenance burden.

Reply:
@hermet Yes I agree this would be simpler, however I'm not absolutely certain that this trick will guarantee that FMA isn't used on here future compilers or other platforms.
But if we have a test in place to catch this I suppose yes it's the simplest solution.

On another note I suspect this kind of incorrect optimization could affects other parts of the code, we'll need to keep an eye on this.